lib/notification.js

```diff
@@ -0,0 +1,341 @@
+};
+
+Notification.defaultProps = {
+  variant: 'notice',
+  live: 'polite',
+  timeout: 0,
+};
+
+const NotificationToggle = ({ children }) => {
+  const [showNotification, setShow] = React.useState(true);
+  return showNotification ? children(() => setShow(false)) : <Button onClick={() => setShow(true)}>Show Notification</Button>;
+};
+
+export const StoryNotification = () => (
+  <>
+    <p>The Notification component renders an accessible notification.</p>
+    <CodeExample>{`<Notification variant="notice" message="Reconnecting…" />`}</CodeExample>
+    <PropsDefinition>
+      <Prop name="message" required>
+        The text of the notification.
+      </Prop>
+      <Prop name="children">An optional rich-content version of the notification that can include formatting and interactive elements.</Prop>
+      <Prop name="live">The aria-live behavior of the notification: "polite" or "assertive" (default "polite").</Prop>
+      <Prop name="variant">The style of notification: "notice", "success", "error" (default "notice").</Prop>
+      <Prop name="timeout">
+        The time in milliseconds before the notification automatically fades out and calls its "onClose" prop. If no value is provided, the
+        notification will not automatically close.
+      </Prop>
+      <Prop name="onClose">
+        A callback function, called when the close button is clicked or the notification times out. If no value is provided, the notification will not
+        render a close button.
+      </Prop>
+    </PropsDefinition>
+    <h3>variant="notice"</h3>
+    <NotificationToggle>{(onClose) => <Notification variant="notice" onClose={onClose} message="Reconnecting…" />}</NotificationToggle>
+
+    <h3>variant="success"</h3>
+    <NotificationToggle>
+      {(onClose) => (
+        <Notification variant="success" onClose={onClose} message="Added power-passenger to collection linen-collection">
+          <p style={{ marginTop: 0, marginBottom: 'var(--space-1)' }}>Added power-passenger to collection linen-collection</p>
+          <Button as="a" variant="secondary" size="tiny" href="https://glitch.com/@modernserf/linen-collection" rel="noopener noreferrer">
+            Take me there
+          </Button>
+        </Notification>
+      )}
+    </NotificationToggle>
+    <h3>variant="error"</h3>
+    <NotificationToggle>
+      {(onClose) => <Notification variant="error" onClose={onClose} message="Something went wrong. Try refreshing?" />}
+    </NotificationToggle>
+  </>
+);
+
+let currentID = 0;
+const uniqueID = (prefix) => {
+  currentID += 1;
+  return `${prefix}-${currentID}`;
+};
+
+const NotificationContext = React.createContext();
+export const NotificationsConsumer = NotificationContext.Consumer;
+export const useNotifications = () => React.useContext(NotificationContext);
+
+const NotificationsContainer = styled.div`
+  position: fixed;
+  z-index: var(--z-notifications);
+  top: 0;
+  right: 0;
+  width: 18rem;
+  padding: var(--space-1);
+`;
```

Comment:
I agree with Edwin that it would be nice for our notifications to have a max width, and maybe even a standard width. Could we consider including this div by default in a Notification?

Reply:
width and default close behavior are provided by NotificationContainer

Reply:
see https://github.com/FogCreek/shared-components/pull/30#discussion_r312594833

Reply:
Yeah I'm proposing that the top-level exported component be NotificationsContainer, with a pre-assigned width for consistency across community/editor, and the notification as children of the NotificationsContainer component. But this is ok for now, we can just make sure that we coordinate with community to ensure we're using the same widths.

Reply:
NotificationsContainer is exported as part of NotificationsProvider

Reply:
Do you _also_ want NotificationsContainer exported? I figure that notifications are mostly useful within a system -- the Notification component is designed to be used with 
`createNotification`, not so much as a free-standing component.

Reply:
Hmm ok, I've been thinking about this slightly differently because we currently use Observables all over the place to control whether or not notifications show, which unless I'm mistaken wouldn't require the NotificationsProvider, but this is a good point at which to move away from that approach and use something more functional like createNotification. In which case this is probably fine as is.

Reply:
I don't think I'm following how this works though. In the code (not the example in the documentation) I see that you created a component called Content, which contains the notifications and their means of being triggered but it's not clear to me what is being imported and what is being created by the end user. Could you change the documented example to show exactly how this NotificationsProvider component gets used in the context of a system, rather than the isolated snippets right now? We can talk through this on Monday as well if it would be easier.

Reply:
`Notification`, `useNotifications` and `NotificationsProvider` are provided by shared components; the actual content of the notifications is done by the consumer. `Content` is a separate component because `useNotifications` consumes the context provided by `NotificationsProvider` and therefore must be used in a child component. 

In community, our root component looks something like

```js
const App = () => (
  <APIProvider>
    <CurrentUserProvider>
      <NotificationsProvider>
         {/* many layers */}
                          <AppContent />
         {/* many layers */}
      </NotificationsProvider>
    </CurrentUserProvider>
  </APIProvider>
)
```

And we have components defined for our common notifications, which look something like

```js
const AddedProjectToCollectionNotification = ({ project, collection, ...props }) => (
  <Notification 
    variant="success"  
    message={`Added ${project.domain} to collection ${collection.name}`}
    {...props}
  >
    <p >Added {project.domain} to collection {collection.name}</p>
    <Button as="a" variant="secondary" size="tiny" href={collection.fullUrl}>
      Take me there
    </Button>
  </Notification>
)
```

 Which we use in the app like

```js
const AddCollectionProjectPopover = ({ collection }) => {
  const api = useAPI();
  const { createNotification, createErrorNotification } = useNotifications();

  const addProjectToCollection = project => {
    api
      .addProjectToCollection({ project, collection })
      .then(
        () =>
          createNotification(props => (
            <AddedProjectToCollectionNotification
              project={project}
              collection={collection}
              {...props}
            />
          )),
        () => createErrorNotification("Could not add project to collection")
      );
  };

  // ...
};
```